tests/end2end/features/test_focustools_bdd.py

```diff
@@ -0,0 +1,44 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2018 Jay Kamat <jaygkamat@gmail.com>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+import pytest
+import logging
+
+import pytest_bdd as bdd
+bdd.scenaros('focustools.feature')
+
+
+@bdd.then(bdd.parsers.parse("no element should be focused"))
+def check_not_focused(quteproc):
+    """Make sure the completion model was set to something."""
+    quteproc.send_cmd(':jseval document.activeElement == document.body',
+                      escape=False)
+    message = quteproc.wait_for(loglevel=logging.INFO, category='message',
```

Comment:
Is there a better way of getting a value from js back into the tests? I couldn't find anything from a quick search, but I could have missed something...

Reply:
You can `console.log` something in JS and then use `Then the javascript message "..." should be logged`.